### DISCUSSION TITLE
Validate formats in `check_schema`

### DISCUSSION THREAD

**Discussion by alliefitter** [python-jsonschema/jsonschema#1280] [Q&A]
I'm wondering if there's a way to validate formats when using `check_schema`. It seems to allow invalid format names. The below example passes validation, even though `datetime` should be `date-time`. I noticed the concrete implementation of `check_schema` has a `format_checker` argument (even though the `FormatChecker` protocol does not have this argument). Setting the format checker doesn't validate format names either. Is there anyway to get `check_schema` to fail for invalid formats?

```
{
  "$schema": "https://json-schema.org/draft/2020-12/schema",
  "type": "object",
  "properties": {
    "foo": {"type": "string", "minLength": 2},
    "bar": {"type": "number", "minimum": 10},
    "baz": {"type": "string", "format": "datetime"}
  }
}
```

**Comment by Julian** [comment]
Hey there.

Those actually aren't invalid formats according to the spec unfortunately.

Specifically, implementations are supposed to ignore formats that aren't known to them. (See e.g. https://json-schema.org/draft/2020-12/json-schema-validation#name-implementation-requirements)

The general answer to this I still consult is "a schema is not a replacement for your tests", so you should notice this by having a test you can't fail. But if you persist, you can also write a metaschema which constrains the set of formats to just the ones known by the library and then use that to validate your schema as well.
